### ISSUE TITLE
Suppress all gRPC Span Events; suppress all Span Events

### ISSUE THREAD

**Issue by fujin** [open-telemetry/opentelemetry-java-instrumentation#6512]
**Is your feature request related to a problem? Please describe.**
My distributed tracing vendor quota is consumed by span events attached to my spans (e.g. for both the span, and each event as an "event" - we have X events per month).

**Describe the solution you'd like**
I would like to suppress the span events in the gRPC instrumentation. I might also like to suppress all auto-instrumented span events globally or enable them on a case-by-case basis.

**Describe alternatives you've considered**
- Dropping span events with experimental/closed PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/5716
- Testing experimental transform processor described here https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7151#issuecomment-1204292185 https://github.com/gregoryfranklin/opentelemetry-collector-contrib/commit/cc4adfbb822a3b2d7e70ed4b59e57b5b6bc9460f
- Disabling gRPC instrumentation

**Comment by trask** [comment]
hi @fujin, one way to do this would be in a [Java agent extension](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension#extensions) using a wrapping SpanExporter, registered via https://github.com/open-telemetry/opentelemetry-java/blob/bebf684436b0f54c0c81b19eba1dd9d12ff37532/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizer.java#L61

**Comment by fujin** [comment]
Yeah, we certainly could build one and package our own image+extension for usage with opentelemetry operator instrumentation injection. However, I would still prefer to disable by default / disable all span events without any additional work.

I ended up using a filter:
```
      filter/ottl:
        traces:
          spanevent:
            - 'name == "message"'
            - 'name == "ss"'
            - 'name == "cr"'
```

**Comment by trask** [comment]
> I would still prefer to disable by default / disable all span events without any additional work

is this not possible using the opentelemetry collector spanevent filter?

**Comment by fujin** [comment]
Yes of course, that's the filter that I mentioned in my previous reply; well, I am currently specifying each one to drop (by name). I would still prefer to not emit them by default/globally rather than discover a new one and filter it (potentially after eating event budget at my tracing vendor).

**Comment by trask** [comment]
I think it would be worth making this request in the collector-contrib repo where that spanevent filter is maintained

**Comment by fujin** [comment]
hmm.. I'm not sure that I agree. Two different problems/solutions, no? 

The code in this repository is responsible for emitting the span events that I would like my collector not to receive.

The collector configuration that filters those events is acceptable to us but doesn't allowing control over if they're emitted in the first place (without disabling that instrumentation library wholesale). I'd like to keep the instrumentation library, but not the events, without having to rely on a filter. I'm fine with closing it, but I would not say those are the same.

**Comment by trask** [comment]
This is being worked on at the specification level now through a combination of

* https://github.com/open-telemetry/opentelemetry-specification/blob/main/oteps/0265-event-vision.md
* https://github.com/open-telemetry/opentelemetry-configuration/pull/140
* https://github.com/open-telemetry/opentelemetry-specification/pull/4290

(and some followups will be needed still)
